### ISSUE TITLE
no bounding boxes using model trained on linux

### ISSUE THREAD

**Issue by bryantclcqq** [AlexeyAB/darknet#143]
the model.weights trained on linux works well on linux darknet framework. however, when use it on this windows framework, no.bounding boxes shown!  why?  is it possible to apply the model trained on linux to this windows  platform?

**Comment by AlexeyAB** [comment]
Hi, 
* Can you show your cfg-file?
* What command do you use on Windows?
* Can you successfully use `yolo.weights` or `yolo-voc.weights` on the Windows?

**Comment by bryantclcqq** [comment]
1. cfg file is almost the same with yolo_voc.cfg(modify it for training)  ;  
2. use darknet.exe detector demo 
3.yes  yolo.weights works well in windows framework
4. wondering any difference between win and linux for .weights files?

**Comment by bryantclcqq** [comment]
1.cfg file is almost the same with yolo_voc.cfg(modify it for training) ;
2. use darknet.exe detector demo
3. yes yolo.weights works well in windows framework
4. wondering any difference between win and linux for .weights files?

**Comment by ifangcheng** [comment]
I  got the problem  with the i.cfg file as follows:
(command:   darknet.exe detector demo  data\i.data  i.cfg  video.mp4)
########################################################

[net]
# Testing
#batch=1
#subdivisions=1
# Training
batch=64
subdivisions=16
height=416
width=416
channels=3
momentum=0.9
decay=0.0005
angle=0
saturation = 1.5
exposure = 1.5
hue=.1

learning_rate=0.001
burn_in=1000
max_batches = 80200
policy=steps
steps=40000,60000
scales=.1,.1

[convolutional]
batch_normalize=1
filters=32
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=64
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=128
size=3
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=64
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=128
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=256
size=3
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=128
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=256
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=512
size=3
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=256
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=512
size=3
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=256
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=512
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=1024
size=3
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=512
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=1024
size=3
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=512
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=1024
size=3
stride=1
pad=1
activation=leaky


#######

[convolutional]
batch_normalize=1
size=3
stride=1
pad=1
filters=1024
activation=leaky

[convolutional]
batch_normalize=1
size=3
stride=1
pad=1
filters=1024
activation=leaky

[route]
layers=-9

[convolutional]
batch_normalize=1
size=1
stride=1
pad=1
filters=64
activation=leaky

[reorg]
stride=2

[route]
layers=-1,-4

[convolutional]
batch_normalize=1
size=3
stride=1
pad=1
filters=1024
activation=leaky

[convolutional]
size=1
stride=1
pad=1
filters=35
activation=linear


[region]
anchors =  1.3221, 1.73145, 3.19275, 4.00944, 5.05587, 8.09892, 9.47112, 4.84053, 11.2364, 10.0071
bias_match=1
classes=2
coords=4
num=5
softmax=1
jitter=.3
rescore=1

object_scale=5
noobject_scale=1
class_scale=1
coord_scale=1

absolute=1
thresh = .6
random=1

**Comment by ifangcheng** [comment]
I got the problem with the i.cfg file as follows:
(command: darknet.exe detector demo data\i.data i.cfg video.mp4)
########################################################

[net]

#Testing

#batch=1
#subdivisions=1

#Training

batch=64
subdivisions=16
height=416
width=416
channels=3
momentum=0.9
decay=0.0005
angle=0
saturation = 1.5
exposure = 1.5
hue=.1

learning_rate=0.001
burn_in=1000
max_batches = 80200
policy=steps
steps=40000,60000
scales=.1,.1

[convolutional]
batch_normalize=1
filters=32
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=64
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=128
size=3
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=64
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=128
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=256
size=3
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=128
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=256
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=512
size=3
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=256
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=512
size=3
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=256
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=512
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=1024
size=3
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=512
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=1024
size=3
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=512
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=1024
size=3
stride=1
pad=1
activation=leaky

#######

[convolutional]
batch_normalize=1
size=3
stride=1
pad=1
filters=1024
activation=leaky

[convolutional]
batch_normalize=1
size=3
stride=1
pad=1
filters=1024
activation=leaky

[route]
layers=-9

[convolutional]
batch_normalize=1
size=1
stride=1
pad=1
filters=64
activation=leaky

[reorg]
stride=2

[route]
layers=-1,-4

[convolutional]
batch_normalize=1
size=3
stride=1
pad=1
filters=1024
activation=leaky

[convolutional]
size=1
stride=1
pad=1
filters=35
activation=linear

[region]
anchors = 1.3221, 1.73145, 3.19275, 4.00944, 5.05587, 8.09892, 9.47112, 4.84053, 11.2364, 10.0071
bias_match=1
classes=2
coords=4
num=5
softmax=1
jitter=.3
rescore=1

object_scale=5
noobject_scale=1
class_scale=1
coord_scale=1

absolute=1
thresh = .6
random=1

**Comment by AlexeyAB** [comment]
@bryantclcqq

> 1.cfg file is almost the same with yolo_voc.cfg(modify it for training) ;
> 2. use darknet.exe detector demo
> 3. yes yolo.weights works well in windows framework
> 4. wondering any difference between win and linux for .weights files?

If the weight file was different, then the default `yolo.weight` would also not work, but `yolo.weight` is works.
* Do you use this command `darknet.exe detector test data/obj.data obj.cfg obj.weights dog.jpg`?
* and this command line `darknet.exe detector demo data/obj.data obj.cfg obj.weights test.mp4` ?

You can upload your 4 files: data, names, cfg and weights, so I'll try to check, does it works.

**Comment by AlexeyAB** [comment]
@ifangcheng 

> I got the problem with the i.cfg file as follows:
> (command: darknet.exe detector demo data\i.data i.cfg video.mp4)

1. You should add `i.weights` to your command line: `darknet.exe detector demo data\i.data i.cfg i.weights video.mp4`

2. Also if you train model on Windows, then you should base your .cfg-file on `yolo-voc.2.0.cfg` instead of `yolo-voc.cfg`

Base your cfg on this file `yolo-voc.2.0.cfg`: https://github.com/AlexeyAB/darknet/blob/master/build/darknet/x64/yolo-voc.2.0.cfg

As mentoid here: https://github.com/AlexeyAB/darknet#how-to-train-to-detect-your-custom-objects

**Comment by bryantclcqq** [comment]
1.sorry，I cannot upload the files.
2. the good news is that   after i retrain the model under windows framework,  it now can show the bounding boxes,even though i used the yolo_voc.cfg instead of  yolo_voc_2.cfg
3. but, still i am not sure  is there any one successfully run the model trained on linux?

**Comment by AlexeyAB** [comment]
@bryantclcqq All the models that I trained on Linux, worked successfully on Windows. But for this I used old commits, some monthes ago.

**Comment by ifangcheng** [comment]
@AlexeyAB
many thx for reply！

“Also if you train model on Windows, then you should base your .cfg-file on yolo-voc.2.0.cfg instead of yolo-voc.cfg”
——   what if I train model on linux?  which one should I use ?  yolo-voc.2.0.cfg or yolo-voc.cfg?

**Comment by AlexeyAB** [comment]
@ifangcheng 
If you use original Linux fork, then you can use any cfg-file. Joseph periodically adds new options for training to the current yolo-voc.cfg file, so my fork doesn't support them. But models which based on yolo-voc.cfg and trained on Linux can be used on Windows for detection.

* `yolo-voc.2.0.cfg` - is stable not changeable cfg file
* `yolo-voc.cfg` - is current often changeable cfg file
